coverage/artifacts/local_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package artifacts
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"testing"
+
+	"k8s.io/test-infra/coverage/logUtil"
+	"k8s.io/test-infra/coverage/test"
+	"os"
+)
+
+const (
+	covTargetRootRel = "testTarget"
+)
+
+// generates coverage profile by running go test on target package
+func TestProfiling(t *testing.T) {
+	if os.Getenv("GOPATH") != "" {
```

Comment:
This will succeed without testing anything if GOPATH is empty. Is it intended?

Reply:
yes, this test should not be running if GOPATH is empty

Reply:
Ok. Please consider logging this situation so whoever touches your code will be aware of that.

Reply:
why is that true? note: the `GOPATH` env is not required since a few releases ago (I think go 1.9?), `import go/build`, `build.Default.GOPATH` is probably a better way to get this value. (in bash `go env GOPATH`)

Reply:
it also looks like we don't actually use GOPATH...?

Reply:
I checked GOPATH here actually to detect whether it is running by prow here. I don't know why it fails the following test when running on prow. However, when testing locally with either Bazel or go test. There is no issues.
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/test-infra/9644/pull-test-infra-bazel/20490/

Reply:
@BenTheElder

Reply:
that is a very bad assumption. GOPATH is set by podutils jobs on prow which are the replacement for jenkins/bootstrap.py.

Reply:
If you want to know if it is running on prow, there should be an explicit argument provided in CI

Reply:
Needing to know that you're running on `prow` is very concerning. That should not ever be the case. Why do you need to know?

Reply:
I made some changes in this package. It should have resolved the issue. Please see the new implementation @BenTheElder @stevekuznetsov